<?xml version="1.0"?>
<launch>

    <!-- To visualize the path and intermediate pose -->
    <arg name="rviz" default="false"/>

    <node pkg="smb_mission_planner" type="hilti_mock_modules.py"  name="hilti_mock_modules" output="screen">
      <param name="path_topic_name"                   value="/mabi_highlevel_controller/path_direct"/>
      <param name="nav_goal_topic"                    value="/move_base_simple/goal"/>
      <param name="base_odom_topic"                   value="/base_pose_measured"/>
      <param name="ee_goal_topic"                     value="/pose_topic_name"/>
      <param name="controller_manager_namespace"      value="/smb_highlevel_controller"/>
      <param name="hal_optimization_service_name"     value="/high_accuracy_localize"/>
      <param name="hal_data_collection_service_name"  value="/hal_take_measurement"/>
      <param name="hal_update_topic_name"             value="/building_task_manager/task_offset_target_pose"/>
      <param name="confusor_service_name"             value="/to_define"/>
    </node>

    <node pkg="smb_mission_planner" type="path_generator.py"  name="path_generator" output="screen">
      <param name="path_topic_name" value="/mabi_highlevel_controller/path_direct"/>
      <param name="x0"              value="4.0"/>
      <param name="y0"              value="4.0"/>
      <param name="z0"              value="1.0"/>
      <param name="yaw_deg"         value="45"/>
    </node>

    <group if="$(arg rviz)">
        <node name="rviz" type="rviz" pkg="rviz" args="-d $(find smb_mission_planner)/configs/hilti_demo/config.rviz" output="screen"/>
    </group>

</launch>